### PR TITLE
Fix 6181/6182

### PR DESCRIPTION
**Pull request by half-life666**
# What problem are we solving?
https://github.com/seaweedfs/seaweedfs/issues/6181
https://github.com/seaweedfs/seaweedfs/issues/6182

# How are we solving the problem?
For 6181:
- when no entry is found (`processedTsNs` is 0), try next day

For 6182:
- set larger buf size

# How is the PR tested?
Using weed s3 and transfer 100 * 512MB files, no events is lost

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

**Comment by half-life666** [comment]
The UT failed at command_ec_encode.go, which is fixed in pr 6169 https://github.com/seaweedfs/seaweedfs/pull/6169/commits/9f7588eda5abeefd71c4ccdca378ff2c3fde06f2

File: weed/filer/filer_notify_read.go [hunk 1]
```diff
@@ -39,6 +40,19 @@ func (f *Filer) collectPersistedLogBuffer(startPosition log_buffer.MessagePositi
 
 }
 
+func (f *Filer) HasPersistedLogFiles(startPosition log_buffer.MessagePosition) (bool, error) {
+	startDate := fmt.Sprintf("%04d-%02d-%02d", startPosition.Year(), startPosition.Month(), startPosition.Day())
+	dayEntries, _, listDayErr := f.ListDirectoryEntries(context.Background(), SystemLogDir, startDate, true, math.MaxInt32, "", "", "")
```
Comment (chrislusf) [inline weed/filer/filer_notify_read.go]:
```suggestion
	dayEntries, _, listDayErr := f.ListDirectoryEntries(context.Background(), SystemLogDir, startDate, true, 1, "", "", "")
```

File: weed/server/filer_grpc_server_sub_meta.go [hunk 2]
```diff
@@ -62,8 +63,19 @@ func (fs *FilerServer) SubscribeMetadata(req *filer_pb.SubscribeMetadataRequest,
 			return nil
 		}
 
+		glog.V(4).Infof("processed to %v: %v", clientName, processedTsNs)
 		if processedTsNs != 0 {
 			lastReadTime = log_buffer.NewMessagePosition(processedTsNs, -2)
+		} else {
+			nextDayTs := util.GetNextDayTsNano(lastReadTime.UnixNano())
+			position := log_buffer.NewMessagePosition(nextDayTs, -2)
+			found, err := fs.filer.HasPersistedLogFiles(position)
```
Comment (chrislusf) [inline weed/server/filer_grpc_server_sub_meta.go:72]: this only check whether there are logs in the next day.

What if there are multiple days without any logs?

**Comment by chrislusf** [comment]
the problem happens only when the UTC date changes?

File: weed/server/filer_grpc_server_sub_meta.go (hunk 2 above)
Comment (half-life666) [reply weed/server/filer_grpc_server_sub_meta.go:72]: > this only check whether there are logs in the next day.
> 
> What if there are multiple days without any logs?

Right, we also need to consider this.

**Comment by half-life666** [comment]
> the problem happens only when the UTC date changes?

Yes.

File: weed/server/filer_grpc_server_sub_meta.go (hunk 2 above)
Comment (half-life666) [reply weed/server/filer_grpc_server_sub_meta.go:72]: If there are multiple days without any logs, the for loop should loop every day, until today? I can verify this in our teststand.

Comment (half-life666) [reply weed/server/filer_grpc_server_sub_meta.go:72]: Tested in our env that, last weekend there is no IO for 2 days, today we write some files and new events can still be subscribed with no problems.

**Comment by zemul** [comment]
For the same problem, the file uploaded through the filer can be displayed on mount, but the file uploaded through s3 cannot be displayed on mount (it can be seen after restart mount). This pr does not seem to have specific repair details.

**Comment by zemul** [comment]
> For the same problem, the file uploaded through the filer can be displayed on mount, but the file uploaded through s3 cannot be displayed on mount (it can be seen after restart mount). This pr does not seem to have specific repair details.对于相同的问题，通过文件管理器上传的文件可以在挂载时显示，但通过 s3 上传的文件无法在挂载时显示（重新启动挂载后可以看到）。这个 pr 似乎没有具体的维修细节。


Can't be repeated, sometimes normal.(Small file,not related to part upload.)

**Pull request merged**